app_dart/lib/src/request_handlers/dart_internal_subscription.dart

```diff
@@ -2,31 +2,208 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:cocoon_service/src/request_handling/body.dart';
+import 'package:cocoon_service/src/model/luci/buildbucket.dart';
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
 import 'package:meta/meta.dart';
+import 'package:retry/retry.dart';
 
 import '../../cocoon_service.dart';
+import '../model/appengine/task.dart';
+import '../model/appengine/commit.dart';
+import '../request_handling/exceptions.dart';
 import '../request_handling/subscription_handler.dart';
+import '../service/datastore.dart';
+import '../service/exceptions.dart';
 import '../service/logging.dart';
 
+/// TODO(drewroengoogle): Make this subscription generic so we can accept more
+/// than just dart-internal builds.
+///
 /// An endpoint for listening to build updates for dart-internal builds and
 /// saving the results to the datastore.
 ///
 /// The PubSub subscription is set up here:
 /// https://console.cloud.google.com/cloudpubsub/subscription/detail/dart-internal-build-results-sub?project=flutter-dashboard
 @immutable
 class DartInternalSubscription extends SubscriptionHandler {
-  /// Creates an endpoint for listening to dart-internal build results.
+  /// Creates an endpoint for listening for dart-internal build results.
+  /// The message should contain a single buildbucket id
   const DartInternalSubscription({
     required super.cache,
     required super.config,
     super.authProvider,
+    required this.buildBucketClient,
+    @visibleForTesting this.datastoreProvider = DatastoreService.defaultProvider,
+    this.retryOptions = Config.buildbucketRetry,
   }) : super(subscriptionName: 'dart-internal-build-results-sub');
 
+  final BuildBucketClient buildBucketClient;
+  final DatastoreServiceProvider datastoreProvider;
+  final RetryOptions retryOptions;
+
   @override
   Future<Body> post() async {
-    log.fine('Consumed message: ${message.data}');
+    final DatastoreService datastore = datastoreProvider(config.db);
+
+    log.info("Converting message data to non nullable int");
+
+    final int buildbucketId = int.parse(message.data.toString().replaceAll("'", ""));
```

Comment:
Should we pass the messages as json? that way we don't need to do this adhoc parsing.